packages/diracx-web-components/src/components/JobMonitor/JobSearchBar.tsx

```diff
@@ -0,0 +1,217 @@
+      setFilters={setFilters}
+      createSuggestions={(
+        previousToken: SearchBarToken | undefined,
+        previousEquation: SearchBarTokenEquation | undefined,
+        equationIndex?: number,
+      ) =>
+        createSuggestions(
+          diracxUrl,
+          accessToken,
+          previousToken,
+          previousEquation,
+          columns,
+          searchBody,
+          equationIndex,
+        )
+      }
+      allowKeyWordSearch={false} // Disable keyword search for job monitor
+    />
+  );
+}
+
+/**
+ * Creates suggestions for the search bar based on the current tokens
+ * If necessary, it fetches job summaries from the server to get personalized suggestions
+ *
+ * @param diracxUrl The URL of the DiracX server.
+ * @param accessToken The access token for authentication, which can be undefined if not authenticated.
+ * @param previousToken The previous token, which can be undefined if no token is focused.
+ * @param previousEquation The previous equation, which can be undefined if no equation is focused.
+ * @param columns The columns to be used for suggestions, which are used to determine the categories and types.
+ * @param searchBody The search body to be sent along with the request (optional).
+ * @param searchBodyIndex The index of the search body, which is used to determine the current search context (optional).
+ * @returns A list of suggestions based on the current tokens and data.
+ */
+async function createSuggestions(
+  diracxUrl: string | null,
+  accessToken: string | undefined,
+  previousToken: SearchBarToken | undefined,
+  previousEquation: SearchBarTokenEquation | undefined,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  columns: ColumnDef<Job, any>[],
+  searchBody: SearchBody,
+  searchBodyIndex?: number,
+): Promise<SearchBarSuggestions> {
+  let data: JobSummary[] = [];
+
+  const search = [...(searchBody?.search || [])];
+
+  const newSearchBody = {
+    ...searchBody,
+    search: search.slice(0, searchBodyIndex),
+  };
+
+  const fetchJobSummary = async (category: string) => {
+    if (diracxUrl && accessToken) {
+      try {
+        const result = await getJobSummary(
+          diracxUrl,
+          [category],
+          accessToken,
+          newSearchBody,
+        );
+        data = result.data || [];
+      } catch {
+        throw new Error("Failed to fetch job summary");
+      }
+    }
+  };
+
+  if (
+    !previousToken ||
+    !previousEquation ||
+    previousEquation.items.length === 0 ||
+    previousToken.type.startsWith("custom") ||
+    previousToken.type === "value"
+  ) {
+    const items = columns.map((column) => column.header as string);
+    const types = columns.map(
+      (column) => `category_${column.meta?.type || "string"}`,
+    );
+
+    return {
+      items: items,
+      type: types,
+    };
+  }
+
+  // Here, we need personalized suggestions based on the previous token
+  if (previousToken.type === "operator_string") {
```

Comment:
I am not entirely sure to understand all the possible values of `previousToken.type`. Shouldn't it be an enum? And shouldn't it be splitted into 2 different parameters? It sounds like you have something like `<nature>_<type>`, but may be I am not understanding it correctly.